roles/testnode/files/libexec/smart.sh

```diff
@@ -0,0 +1,254 @@
+#!/bin/bash
+# Description:	Bash script to check drive health using pending, uncorrectable,
+# and reallocated sector count
+#
+# Nagios return codes: 0 = OK; 1 = WARNING; 2 = CRITICAL; 3 = UNKNOWN
+# SMART Attribute Codes:
+#   5 = Reallocated
+#   187 = Reported Uncorrect
+#   197 = Pending
+#   198 = Uncorrectable Sector Count
+#
+# TO-DO: Add support for dynamic SMART attribute lookup.  For example,
+#        187 is reported for Seagate HDD and all SSDs but not Hitachi HDDs.
+#
+# See https://en.wikipedia.org/wiki/S.M.A.R.T.#ATA_S.M.A.R.T._attributes
+
+
+### Pre-flight checks ###
+# Set raid var then check for cli64 command and bail if missing
+if lspci | grep -qi areca
+then
+  raid=true
+else
+  raid=false
+fi
+
+if ! [ -x "$(command -v cli64)" ] && [ "$raid" = true ]
+then
+  echo "ERROR - cli64 command not found or is not executable"
+  exit 3
+fi
+
+# Check for smartmontools and bail if missing
+if ! [ -x "$(command -v smartctl)" ]
+then
+  echo "ERROR - smartctl is not installed or is not executable"
+  echo "yum/apt-get install smartmontools"
+  exit 3
+fi
+
+### Variables ###
+numdrives=0 # total number of drives (or RAID slots) discovered
+failingdrives=0 # Number of failed, failing, and/or missing drives
+rc=0 # Return code for nagios
+unknownmsg="Unknown error" # Fallback message for UNKNOWN return code output
+
+# Array of messages indicating drive health.  Output after nagios status.
+declare -a messages
+
+### Functions ###
+
+# Gather smart data for drives behind Areca RAID controller
+# NOTE: Only works when individual disks set to JBOD for now
+areca_smart ()
+{
+  numdrives=$(sudo cli64 disk info | grep -E "Slot#[[:digit:]]" | wc -l)
+  # Loop through all disks not marked as 'N.A.' or 'Failed'
+  for slot in $(sudo cli64 disk info | grep -E "Slot#[[:digit:]]" \
```

Comment:
Yeah I knew I was unnecessarily calling cli64 more than needed.  Just figured this trick out...

```
# output=$(sudo cli64 disk info)                                                                  
# 
# echo $output
# Enc# Slot# ModelName Capacity Usage =============================================================================== 1 01 Slot#1 ST31500341AS 1500.3GB Raid Set # 000 2 01 Slot#2 ST31500341AS 1500.3GB Raid Set # 000 3 01 Slot#3 ST1500DM003-9YN16G 1500.3GB Raid Set # 000 4 01 Slot#4 ST31500341AS 1500.3GB Raid Set # 000 5 01 Slot#5 ST1500DM003-9YN16G 1500.3GB Free 6 01 Slot#6 ST31500341AS 1500.3GB Raid Set # 000 7 01 Slot#7 N.A. 0.0GB N.A. 8 01 Slot#8 ST1500DM003-9YN16G 1500.3GB Raid Set # 000 =============================================================================== GuiErrMsg<0x00>: Success.
# 
# 
# echo "$output"
  # Enc# Slot#   ModelName                        Capacity  Usage
===============================================================================
  1  01  Slot#1  ST31500341AS                     1500.3GB  Raid Set # 000  
  2  01  Slot#2  ST31500341AS                     1500.3GB  Raid Set # 000  
  3  01  Slot#3  ST1500DM003-9YN16G               1500.3GB  Raid Set # 000  
  4  01  Slot#4  ST31500341AS                     1500.3GB  Raid Set # 000  
  5  01  Slot#5  ST1500DM003-9YN16G               1500.3GB  Free      
  6  01  Slot#6  ST31500341AS                     1500.3GB  Raid Set # 000  
  7  01  Slot#7  N.A.                                0.0GB  N.A.      
  8  01  Slot#8  ST1500DM003-9YN16G               1500.3GB  Raid Set # 000  
===============================================================================
GuiErrMsg<0x00>: Success.
```

Fixed in latest revision